build/common.sh

```diff
@@ -244,7 +256,9 @@ fi
 . ${DEVICEDIR}/${PRODUCT_DEVICE_REAL}.conf
 
 # reload the kernel according to device specifications
-export PRODUCT_KERNEL="${PRODUCT_KERNEL}${PRODUCT_DEVICE+"-${PRODUCT_DEVICE}"}"
+if [ ${PRODUCT_ARCH} != armv6 -a ${PRODUCT_ARCH} != armv7 -a ${PRODUCT_ARCH} != aarch64 ]; then
```

Comment:
actually that was supposed to be an ARM thing, but now it's applied to anything not ARM

This would be more appropriate with a device config and using the following:

https://github.com/opnsense/tools/blob/17fa565a1589d029169504b3414d9aea4532b193/device/A10.conf#L12

Reply:
That should be in RPI3.conf it seems

Reply:
Hi there. There are two reasons that I put these codes in common scripts instead of device config files.
First of all, this is a common configuration for all ARM devices. So I don't want to redundant the codes to every config file.
Moreover, apart from some specific ARM devices, there are also generic ARM machines, such as QEMU and ESXi virtual machines. These machines are not listed in `tools/device`, and I don't think they should.